pandas/tests/frame/test_convert_to.py

```diff
@@ -175,7 +171,108 @@ def test_to_records_index_name(self):
     def test_to_records_with_unicode_index(self):
         # GH13172
         # unicode_literals conflict with to_records
-        result = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a')\
+        result = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a') \
             .to_records()
         expected = np.rec.array([('x', 'y')], dtype=[('a', 'O'), ('b', 'O')])
         tm.assert_almost_equal(result, expected)
+
+    def test_to_records_with_tz_gmt(self):
+        # GH13937
+        df = DataFrame({'datetime': date_range('2016-01-01',
+                                               periods=10,
+                                               freq='S',
+                                               tz='UTC')},
+                       index=date_range('2016-01-01',
+                                        periods=10,
+                                        freq='S',
+                                        tz='UTC'))
+
+        dfgmt = df.tz_convert("GMT")
+        tm.assertIsNot(df.to_records().index[0].tzinfo,
+                       dfgmt.to_records().index[0].tzinfo)
+
+    def test_to_records_with_tz_convert_to_utc(self):
+        # GH13937
+
+        df_gmt = DataFrame(date_range('2016-01-01',
+                                      periods=10,
+                                      freq='S',
+                                      tz='GMT'),
+                           index=date_range('2016-01-01',
+                                            periods=10,
+                                            freq='S',
+                                            tz='GMT'))
+        df_utc = DataFrame(date_range('2016-01-01',
+                                      periods=10,
+                                      freq='S',
+                                      tz='GMT'),
+                           index=date_range('2016-01-01',
+                                            periods=10,
+                                            freq='S',
+                                            tz='UTC'))
+        dfgmt = df_gmt[:].tz_convert("UTC")
+        tm.assert_numpy_array_equal(df_utc.to_records(), dfgmt.to_records())
+
+    def test_to_records_with_tz_as_equal_frame(self):
```

Comment:
try to consolidate these tests into a single test (but with multiple parts is ok), its just confusing to start with 3 different frames, start with 1, then do a 

```
result = ....
expected = ....
tm.assert_numpy_array_equal(....)

# next part
....

# last part
...
```